src/FSharp.Data.GraphQL.Server/Execution.fs

```diff
@@ -286,34 +272,82 @@ module ResolverTree =
                 errorOp path' err
             | ResolverObjectNode node ->
                 let path' = if node.Name <> "__index" then (node.Name :> obj)::path else path
-                let ts = node.Children |> Array.map(helper path')
+                let mapper (c : AsyncVal<ResolverTree>) = asyncVal {
+                    let! c' = c
+                    return helper path' c'
+                }
+                let ts = node.Children |> Array.map mapper
                 nodeOp path' node.Name node.Value ts
             | ResolverListNode node ->
                 let path' = (node.Name :> obj)::path
-                let ts = node.Children |> Array.mapi(fun i c -> helper ((i :> obj)::path') c)
+                let mapper (i : int) (c : AsyncVal<ResolverTree>) = asyncVal {
+                    let! c' = c
+                    return helper ((box i)::path') c'
+                }
+                let ts = node.Children |> Array.mapi mapper
                 listOp path' node.Name node.Value ts
         helper []
 
+let private foldChildren (children : AsyncVal<AsyncVal<KeyValuePair<string, obj> * Error list>> []) =
```

Comment:
I don't like this folding design, but it works as expected. I think I will try to make a better design in future.